core/src/test/java/com/linecorp/armeria/server/AnnotatedHttpServiceExceptionHandlerTest.java

```diff
@@ -0,0 +1,186 @@
+        @ExceptionHandler(AnticipatedExceptionHandler2.class)
+        public HttpResponse asyncHttpResponse(ServiceRequestContext ctx, HttpRequest req) {
+            return HttpResponse.from(asyncStage());
+        }
+
+        private CompletionStage<HttpResponse> stage() {
+            throw new AnticipatedException("Oops!");
+        }
+
+        private <T> CompletionStage<T> asyncStage() {
+            final CompletableFuture<T> future = new CompletableFuture<>();
+            // Execute 100 ms later.
+            Executors.newSingleThreadScheduledExecutor().schedule(
+                    () -> future.completeExceptionally(new AnticipatedException("Oops!")),
+                    100, TimeUnit.MILLISECONDS);
+            return future;
+        }
+    }
+
+    // No exception handler is specified.
+    @Converter(target = String.class, value = UnformattedStringConverter.class)
+    public static class MyService2 {
+
+        @Get("/sync")
+        public String sync(ServiceRequestContext ctx, HttpRequest req) {
+            throw new IllegalArgumentException("Oops!");
+        }
+    }
+
+    @Converter(target = String.class, value = UnformattedStringConverter.class)
+    @ExceptionHandler(BadExceptionHandler.class)
+    public static class MyService3 {
+
+        @Get("/asyncStage")
+        public HttpResponse asyncHttpResponse(ServiceRequestContext ctx, HttpRequest req) {
+            return HttpResponse.from(asyncStage());
+        }
+
+        private <T> CompletionStage<T> asyncStage() {
+            final CompletableFuture<T> future = new CompletableFuture<>();
+            // Execute 100 ms later.
+            Executors.newSingleThreadScheduledExecutor().schedule(
+                    () -> future.completeExceptionally(new AnticipatedException("Oops!")),
+                    100, TimeUnit.MILLISECONDS);
+            return future;
+        }
+    }
+
+    static class AnticipatedExceptionHandler1 implements ExceptionHandlerFunction {
+        @Override
+        public HttpResponse handle(RequestContext ctx, HttpRequest req, Throwable cause) {
+            return HttpResponse.of(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, "handler1");
+        }
+    }
+
+    static class AnticipatedExceptionHandler2 implements ExceptionHandlerFunction {
+        @Override
+        public HttpResponse handle(RequestContext ctx, HttpRequest req, Throwable cause) {
+            return HttpResponse.of(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, "handler2");
+        }
+    }
+
+    static class BadExceptionHandler implements ExceptionHandlerFunction {
+        @Override
+        public HttpResponse handle(RequestContext ctx, HttpRequest req, Throwable cause) {
+            final DefaultHttpResponse response = new DefaultHttpResponse();
+            response.write(HttpHeaders.of(HttpStatus.OK));
+            Executors.newSingleThreadScheduledExecutor().schedule(
+                    (Runnable) response::close, 1, TimeUnit.SECONDS);
```

Comment:
- `ctx.eventLoop().schedule(...)`?
- Could use longer delay so that the response is not closed before the assertion in a very slow machine?